pkg/test/helm/commands.go

```diff
@@ -0,0 +1,39 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package helm
+
+import (
+	"fmt"
+
+	"istio.io/istio/pkg/test/shell"
+)
+
+// Template calls "helm template".
+func Template(deploymentName, namespace, chartDir, valuesFile string, s *Settings) (string, error) {
```

Comment:
If the deployment moves to `/pkg/test/kube/helm` as I suggested above, this should go there too.

In fact, I feel that this should just be a member of the deployment.

Reply:
I'd prefer to keep utility packages separate from higher-level packages.

This helm package is simply wrapping over the helm command-line. It is very self-contained, and can be referenced externally without having to worry about pulling in arbitrary/unrelated things. If someone wants to write automation using Helm, they can simply do that.

Similar to "deployment" package, this actually contained more things, such as init, create/install and delete commands, before I settled on Helm Yaml generation as the current/single way of deploying. These are all closely aligned with the Helm tool.